app/main.py

```diff
@@ -1,2 +1,61 @@
+from typing import Hashable, Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self) -> None:
+        self._capacity = 8
+        self._threshold = int(self._capacity * 2 / 3)
+        self._length = 0
+        self._increase_value = 2
+        self.table = [[] for _ in range(self._capacity)]
+
+    def __len__(self) -> int:
+        return self._length
+
+    def __resize_table(self) -> None:
+
+        table_ = self.table.copy()
+        self._length = 0
+        self._capacity *= 2
+        self._threshold = int(self._capacity * 2 / 3)
+        self.table = [[] for _ in range(self._capacity)]
+
+        for cell in table_:
+            if cell:
+                self[cell[0]] = cell[1]
+
+    def __getitem__(self, item: Hashable) -> list or None:
+        hash_ = hash(item)
+        index = hash_ % self._capacity
+
+        while True:
+            if not self.table[index]:
```

Comment:
If you want, you can move this check to `while` statement, and raise an error upon exit.

Reply:
looks better